test/jdk/java/util/Collections/CheckedMapBash.java

```diff
@@ -183,4 +183,15 @@ public static Collection<Object[]> makeCheckedMaps() {
         };
         return Arrays.asList(params);
     }
+
+    @Test(groups = "type_check")
+    public static void testTypeCheck() {
+        try {
+            Map m = Collections.checkedMap(new HashMap<>(), Integer.class, Integer.class);
+            m.merge("key", "value", (v1, v2) -> null);
+            m.merge("key", 3, (v1, v2) -> v2);
```

Comment:
May be better to seperate these two cases into two tests. Because if the first case fails, the latter can't be verified now.